drf_roles/mixins.py

```diff
@@ -29,8 +29,9 @@ def __init__(self, **_kwargs):
     def _call_role_fn(self, fn, *args, **kwargs):
         """Attempts to call a role-scoped method"""
         try:
-            role_name = self._get_role(self.request.user)
-            role_fn = "{}_for_{}".format(fn, role_name)
+            if not getattr(self, '_role_name', None):
```

Comment:
try using `lru_cache` from functools python module for result cache

https://docs.python.org/3/library/functools.html#functools.lru_cache